pkg/util/deployment/deployment.go

```diff
@@ -327,21 +335,19 @@ func IsPodAvailable(pod *api.Pod, minReadySeconds int32) bool {
 	return false
 }
 
-func GetPodsForReplicaSets(c clientset.Interface, replicaSets []*extensions.ReplicaSet) ([]api.Pod, error) {
+func GetPodsForReplicaSets(replicaSets []*extensions.ReplicaSet, podList *api.PodList) ([]api.Pod, error) {
```

Comment:
WDYT of renaming this something like 'FilterPodsMatchingReplicaSets' since it is now just filtering the list?  Consider using [collection functions such as these](https://gobyexample.com/collection-functions) to do the heavy lifting.